### PR TITLE
May 349 deploy order box

### PR DESCRIPTION
**Pull request by elibar-uk**
### Goals
- **[MAY-349](https://globalpersonals.atlassian.net/browse/MAY-349):** Deploy order box

adds deployment order to project, edits the view - adds select options for deployment order 

***Progress: 90%***

screen shot 2016-12-14 at 15 33 57
---

### CODE REVIEW

File: db/migrate/20150902133758_create_projects.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ def change
     create_table :projects do |t|
       t.integer :deployment_id
       t.integer :branch_id
+      t.integer :deployment_order
```
Comment (mendable) [inline db/migrate/20150902133758_create_projects.rb]: When making a change to an _existing_ database schema we should change the existing database by adding a new migrations, not than editing a previous ones.

When the database already exists, we'd run `rake db:migrate` to apply database changes, which would only run _new_ migrations - it wouldn't re-run old migrations or wipe existing tables/data.

The migrations are a "changelog" of the versions of the database.

Also see: http://guides.rubyonrails.org/v4.1/migrations.html

File: app/views/shared/_project_form.html.haml [hunk 2]
```diff
@@ -8,6 +8,18 @@
...
 
     .form-group
+      %label.col-sm-3.control-label{:for => "deployment_order"} Deployment Order
+      .col-sm-3
+        %select.order_options.form-control{:name => "deployment[projects][deployment_orders][]", :required => "", :placeholder => "Put the order number to deploy"}
+          - if project
+            - project.deployment.projects.each do | each_project |
+              %option{:value => each_project.deployment_order, :selected => each_project.deployment_order == project.deployment_order ? "selected" : nil}= each_project.deployment_order
+          - else
+            %option{:value => "" }
```
Comment (mendable) [inline app/views/shared/_project_form.html.haml]: spacing before ` }`

File: app/views/deployments/new.html.haml [hunk 3]
```diff
@@ -122,12 +122,26 @@
...
+
     // add form
     $('.add-projects').click(function(e){
       e.preventDefault();
       var l = Ladda.create(this);
       l.start();
       $('#app-area').append('#{j render 'shared/project_form', project: nil}');
+      $('#app-area .projects-wrap .order_options').each(function() {
+        var project = $('#app-area .projects-wrap:last')
+        $( this ).append($('<option>', {value:(($('#app-area .projects-wrap')
```
Comment (mendable) [inline app/views/deployments/new.html.haml]: can we be consistent with the use of `$( this )` vs `$(this)` ? Existing code in this file seems to use the unspaced version.

**Comment by nirpius** [comment]
Tests peformed. All passed.
- [x] Can add and remove repos. Largest assignable position value is updated correctly
- [x] If moving from 4 repos to 3 repos, repo marked as "4" changes to "Please Select"
- [x] Can assign positions in any order eg 2,3,1,4
- [x] Can change position after assigning
- [x] Cannot contrive a situation where 2 repos have the same position value
- [x] Cannot submit an RR unless al repos have a position value

**Pull request merged**